core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/MarecoAllowance.java

```diff
@@ -349,8 +350,10 @@ private double computeCoastingBeginSpeed(double initialSpeed,
             res.add(accelerationPart);
         }
         if (capacitySpeedLimit > 0) {
-            var maintainPart = generateMaintainPart(capacitySpeedLimit);
```

Comment:
We cap the base curve rather than adding a constant speed limit so that train stops can work properly, without being limited at 30km/h